Simplenote/src/test/java/com/automattic/simplenote/viewmodels/TagDialogViewModelTest.kt

```diff
@@ -0,0 +1,126 @@
+package com.automattic.simplenote.viewmodels
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.automattic.simplenote.R
+import com.automattic.simplenote.models.Tag
+import com.automattic.simplenote.repositories.TagsRepository
+import com.automattic.simplenote.utils.getLocalRandomStringOfLen
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+
+
+class TagDialogViewModelTest {
```

Comment:
Suggestion (💡): After I have completed the testing solution to the best of my abilities, I usually run the tests with coverage to see whether I missed testing something obvious. In this case the below were missed:
- The `close()` function is untested.
- The `if (currentState == null) { ... }` paths within the `renameTagIfValid()` and `renameTag()` functions are untested.

PS: I would also test the `renameTag()` function individually. I know this function is tested through the `renameTagIfValid()`, but consider what would happen if this parent calling function is, for example, removed at some point. Wdyt?

Reply:
Yes, those were missing tests. Added tests for `close` and `renameTag`. Fixed in e025a750967a98e21f9288b952c826817e6e67ac.